### PR TITLE
introduction of FBP parser and splitting of repo

### PR DESCRIPTION
**Pull request by maschmann**
- introduce FBP dsl parser, rework graph
- shutdown optimizations
- update README to have all info concerning the new functionality from FBP parser
- add example(s) for FBP file loading

### CODE REVIEW

File: examples/linecount/count.php [hunk 1]
```diff
@@ -13,8 +13,10 @@
 // Include standard autoloader
 require __DIR__ . '/../../vendor/autoload.php';
 
+$definition = new \PhpFlo\Fbp\FbpDefinition(['properties' => ['name' => 'linecount']]);
```
Comment (aretecode) [inline examples/linecount/count.php]: mismatching namespaces with $graph below

File: examples/linecount/countFromFbp.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+if (!isset($_SERVER['argv'][1])) {
+    die("You must provide a filename\n");
+}
+$fileName = $_SERVER['argv'][1];
+
+// Include standard autoloader
+require __DIR__ . '/../../vendor/autoload.php';
+
+$builder = new \PhpFlo\Builder\ComponentFactory();
```
Comment (aretecode) [inline examples/linecount/countFromFbp.php]: mismatch here as well

File: lib/PhpFlo/Interaction/Port.php [hunk 3]
```diff
@@ -66,35 +66,51 @@ public function onBeginGroup($groupName, SocketInterface $socket)
...
     }
 
     /**
      * @throws SocketException
      */
     public function connect()
     {
-        if (!$this->socket) {
-            throw new SocketException("No socket available");
+        if (null != $this->socket) {
```
Comment (aretecode) [inline lib/PhpFlo/Interaction/Port.php]: stricter comparison please ftw

File: lib/PhpFlo/Network.php [hunk 4]
```diff
@@ -230,11 +230,27 @@ public function addInitial($data, $node, $port)
...
 
     /**
+     * Cleanup network state after runs.
+     *
      * @return $this
      */
     public function shutdown()
     {
-        //@todo: implement
+        foreach ($this->processes as $process) {
```
Comment (aretecode) [inline lib/PhpFlo/Network.php]: 👍

**Comment by maschmann** [comment]
@aretecode did the requested changes, also rewrote part of the README of [https://github.com/phpflo/phpflo-fbp/blob/master/README.md#fbp-dsl-defintions](phpflo-fbp) according to our discussion on FBP format ;-)

**Pull request merged**